cue/util/iterate_test.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2023 The KubeVela Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util_test
+
+import (
+	"testing"
+
+	"cuelang.org/go/cue"
+	"cuelang.org/go/cue/cuecontext"
+	"github.com/stretchr/testify/require"
+
+	"github.com/kubevela/pkg/cue/util"
+)
+
+func TestIterate(t *testing.T) {
+	value := cuecontext.New().CompileString(`
+	a: ["a", "b", "c"]
+	#x: string
+	b: {
+		c: "val"
+		d: "d"
+	}`)
+	var results []string
+	stop := util.Iterate(value, func(v cue.Value) (stop bool) {
+		if s, err := v.String(); err == nil {
+			results = append(results, s)
+			if s == "val" {
+				return true
+			}
+		}
+		return false
+	})
+	require.Equal(t, []string{"a", "b", "c", "val"}, results)
+	require.True(t, stop)
+}
+
+func TestIterateWithOrder(t *testing.T) {
```

Comment:
Can you add a test case with `if comprehension` like:

```cue
`step1: {}
step2: {prefix: step1.value}
if step2.value > 100 {
	step2_3: {prefix: step2.value}
}
step3: {prefix: step2.value}
step4: {prefix: step3.value}
```

Reply:
ok